### ISSUE TITLE
Bug: when Table A has multiple relations referencing Table B the generated models will not have all of the relations generated

### ISSUE THREAD

**Issue by hadihallak** [APCOvernight/adonis-schema-builder#8]
Example:
-------------------
**posts:**
created_by (references id on the users table)
approved_by (references id on the users table)

**users:**
id

**Expected Outcome:**
-------------------
users model should have two hasMany relations, one for each column

**Actual Outcome:**
---------------------
users model only have one hasMany relation 

--------------
this is most likely caused by the usage of the table name as the method name in the generated model file, so a possible solution is to use a prefix or use the column name as the relation method name when there are more than one  column referencing the same table

**Comment by webdevian** [comment]
Well spotted, do you fancy creating a pull request for this?

I think you would have to implement it in this method - https://github.com/APCOvernight/adonis-schema-builder/blob/master/src/SchemaParser.js#L209

**Comment by hadihallak** [comment]
i made a temporary fix for this issue by using a combination of the table name and the column name instead of just using the table name. however, after trying it i didn't like the outcome as model method names got very messy so i think it might be better to prompt the user to enter the method name using ace instead of the just adding a prefix.

what do you think about using the above mentioned approach?
if you agree with it then maybe i can create a pull request for this next week.

**i also have been editing your package locally and added a couple more features and fixed some other bugs that i might be able to push next week. those includes:**
1. a fully working seeds generator with support for relational data
2. add the option to generate a migration file with only the incremental keys of the tables before generating the migrations. this will help prevent issues when running the migration with relational data heavy applications since we wont have to worry about the order of execution anymore.
3. since you're using .onDelete('set null') for foreign keys i edited the migration generator so that it would delete the .notNullable() for foreign keys since it was causing issues with running the migrations.
4. made the generated factories accept optional data by default so now it will look like: 
```javascript
Factory.blueprint('App/Models/Category', (faker, i ,data) => {
  return {
    name: (data.name) && data.name[i] || faker.name().substring(0, 50)
  }
}),```

**Comment by webdevian** [comment]
Sounds great, I'm looking forward to your PR!

**Comment by webdevian** [comment]
Closed in #9
